src/writer/delete-proposal.ts

```diff
@@ -20,11 +20,14 @@ export async function verify(body): Promise<any> {
 
 export async function action(body): Promise<void> {
   const msg = jsonParse(body.msg);
-  const id = msg.payload.proposal;
+  const proposal = await getProposal(msg.space, msg.payload.proposal);
 
+  const id = msg.payload.proposal;
   const query = `
-  DELETE FROM proposals WHERE id = ? LIMIT 1;
-  DELETE FROM votes WHERE proposal = ?;
+    DELETE FROM proposals WHERE id = ? LIMIT 1;
+    DELETE FROM votes WHERE proposal = ?;
+    UPDATE spaces SET proposal_count = proposal_count - 1, vote_count = vote_count - ? WHERE id = ?;
   `;
-  await db.queryAsync(query, [id, id]);
+
+  await db.queryAsync(query, [id, id, proposal.votes, msg.space]);
```

Comment:
We are increasing `vote_count` on every new vote, but here `proposal.votes` is not guaranteed to be all votes, especially in case of override strategies where `proposal.votes` is updated once in a while. maybe better to get length from `votes` table

Reply:
This is counter is ambiguous. By using true votes count (number of table rows), it will unsync with the results in some case, where we could see 1 vote in the votes list, but 0 votes in the results.

This is an ongoing issue with all the proposal with failed score state.

Reply:
If we unsync the votes with results (which is the best solution for now), we should add a marker next to the votes on UI to let people know the vote is pending, so they know it’s not in the results yet

Reply:
> We are increasing `vote_count` on every new vote, but here `proposal.votes` is not guaranteed to be same as all votes, especially in the case of override strategies where `proposal.votes` is updated once in a while.
> 
> Better to get length from the `votes` table

In which case vote count wouldn't be correct? Even if with override strategy someone may endup having 0 voting power, his vote would still be on the "votes" table. If we take votes length from the table it will be hard to update live, we would need to wait the end of the proposal to do the update which is not ideal.

Reply:
I mean we don't have to worry about results, if this count depends on results, then better to update this count when proposal is closed and not while inserting votes

Reply:
> if we take votes length from the table it will be hard to update live

I mean while deleting proposal, we should reduce based on votes count not with `proposal.votes`

Reply:
wyt? @bonustrack

Reply:
My take would be to keep 2 counters:

- `vote_count`: total number of votes for the proposals, regardless of status
- `(valid|executed|computed|resolved|whatever)_vote_count`: total number of votes with a valid final score (will exclude pending score from override strategies, as well as vote with invalid `score_state` see #280 )

Reply:
Sound complicated, unless I'm missing something it should be rare case to have vote count unsync, and we can always resync / recompute the counter

Reply:
Everything that is not `final` is an unsynced counter:

![Screenshot 2024-04-06 at 14 34 11](https://github.com/snapshot-labs/snapshot-sequencer/assets/495709/b269d44f-86e6-45cc-a517-e996ba93f8ca)

Reply:
> Sound complicated, unless I'm missing something it should be rare case to have vote count unsync, and we can always resync / recompute the counter

An example of unsynced counter: https://github.com/snapshot-labs/snapshot-hub/issues/579

Reply:
Refactored to always use total number of votes